test-data/unit/check-protocols.test

```diff
@@ -2890,6 +2890,22 @@ class SupportsStr(Protocol):
 def ss(s: SupportsStr) -> None: pass
 ss(None)
 
+class SupportsBool(Protocol):
+    def __bool__(self) -> bool: ...
+
+class SupportsBoolLiteralTrue(Protocol):
+    def __bool__(self) -> Literal[True]: ...
+
+class SupportsBoolLiteralFalse(Protocol):
+    def __bool__(self) -> Literal[False]: ...
+
+def sb(s: SupportsBool) -> None: pass
+sb(None)
+def sblt(s: SupportsBoolLiteralTrue) -> None: pass
```

Comment:
I am not sure that this is correct. https://github.com/python/typeshed/blob/ef758b66c0947a73f6225cac0568f3f068cd61d4/stdlib/types.pyi#L627-L629

Reply:
Yes, you're right, it's not correct behavior, but tests passed. How I can fix it? Maybe we have other bug?

Reply:
`None` is quite complicated, because mypy uses pre-`types.NoneType` logic for `None`.
Check out `checkmember` and `NoneType` handling there.

Reply:
I'm implemented check into `visit_none_type` method, because `__bool__` of Protocol not handled by `analyze_none_member_access`